pilot/pkg/model/config.go

```diff
@@ -1062,6 +1088,80 @@ func (store *istioConfigStore) EndUserAuthenticationPolicySpecByDestination(inst
 	return out
 }
 
+// Set jwks_uri through openID discovery if it's not set in auth policy.
+func (store *istioConfigStore) SetAuthenticationPolicyJwksURIs(policy *authn.Policy) {
```

Comment:
May I ask the reason ? Personally I don't have strong preference of calling SetAuthenticationPolicyJwksURIs in authentication.go or config.go; The reason I call SetAuthenticationPolicyJwksURIs in IstioConfigStore is because if AuthenticationPolicyByDestination(method exposed by store interface) will be used in other places in the future and need to set jwks_uri field, it's natural to set in store layer(where cache locates); also it's easy for me to reuse existing test case.

Reply:
I'd like to put all modifier/fallback in to the `GetConsolidateAuthenticationPolicy`, and avoid to use `store.AuthenticationPolicyByDestination` directly (which already the case; there are some fallback logic that prevent us to simply call `AuthenticationPolicyByDestination`). This structure make it more consistent with other store functions, which return CRDs as is (and thus consistent with istio config testing framework). Also, in the future, store implementation could be outside pilot.

Being said that, future code can be moved either way. Let's test if this work first, and iterate on the design before committing more coding.

Reply:
+Joy, could you pls comment from IstioConfigStore side ?